coldfront/core/utils/permissions.py

```diff
@@ -0,0 +1,32 @@
+from functools import wraps
+from django.contrib import messages
+
+def permission_required(permission_check):
+    """
+    Decorator to check if a user has a certain permission before allowing them to access a view.
+    The decorater is used to wrap a test_func from UserPassesTestMixin, which allows gradular refactoring of
+    the permission check logic on test_func without having to change the permission logic.
+
+    :param permission_check: function that returns True if the user has the permission, False otherwise
+    :return:
+    """
+    def decorator(test_func):
```

Comment:
Take multiple as input with `*permission_checks` and then loop over them?

So it'd be:

```python
@permissions_required(perm1, perm2, ..., permK)  # Note the plural
def test_func(self):
    pass
```

As opposed to:

```python
@permission_required(permK)
.
.
.
@permission_required(perm2)
@permission_required(perm1)
def test_func(self):
    pass
```

Not especially opinionated on which is cleaner.

Reply:
I like your suggestion better actually.